File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;
    people.sort((a,b) => a-b)
    console.log(people)
    while (people.length > 0) {
        if (people.length >= 2 && people[0] + people[people.length - 1] <= limit) {
            answer ++;
            people.shift()
            people.pop()
        } else {
            people.pop()
            answer ++
        }
        // console.log(people)
    }
    return answer;
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    people.sort()
    people = deque(people)
    
    while people:
        if len(people) >= 2 and people[-1] + people[0] <= limit:
            answer += 1
            people.pop()
            people.popleft()
        else:
            people.pop()
            answer += 1
    
    return answer